datacache/common.py

```diff
@@ -65,7 +65,7 @@ def build_local_filename(download_url=None, filename=None, decompress=False):
 
     # if no filename provided, use the original filename on the server
     if not filename:
-        digest = hashlib.md5(download_url).hexdigest()
+        digest = hashlib.md5(download_url.encode('utf-8')).hexdigest()
```

Comment:
calling encode() is necessary because md5 now expects a byte buffer instead of a string in Python 3. The choice of UTF-8 is arbitrary though. I'm removing the explicit "utf-8" now, so it'll just default to the user's local encoding, which is I think less confusing.